src/main/java/ch/njol/skript/events/EvtBeacon.java

```diff
@@ -0,0 +1,77 @@
+package ch.njol.skript.events;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptEvent;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import com.destroystokyo.paper.event.block.BeaconEffectEvent;
+import io.papermc.paper.event.block.BeaconActivatedEvent;
+import io.papermc.paper.event.block.BeaconDeactivatedEvent;
+import io.papermc.paper.event.player.PlayerChangeBeaconEffectEvent;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+public class EvtBeacon extends SkriptEvent {
+
+	static {
+		if (Skript.classExists("com.destroystokyo.paper.event.block.BeaconEffectEvent")) {
+			Skript.registerEvent("Beacon Effect", EvtBeacon.class, BeaconEffectEvent.class, "beacon effect")
+				.description("Called when a player gets an effect from a beacon")
+				.examples(
+					"on beacon effect:",
+						"\tbroadcast event-potioneffecttype",
+						"\tbroadcast event-player",
+						"\tbroadcast event-block"
+				)
+				.since("INSERT VERSION")
+				.requiredPlugins("Paper");
+		}
+		if (Skript.classExists("io.papermc.paper.event.block.BeaconActivatedEvent")) {
+			Skript.registerEvent("Beacon Toggle", EvtBeacon.class, new Class[] {BeaconActivatedEvent.class, BeaconDeactivatedEvent.class}, "beacon [:de]activate")
+				.description("Called when a beacon is activated or deactivated")
+				.examples(
+					"on beacon activate:",
+						"\tbroadcast event-block"
+				)
+				.since("INSERT VERSION")
+				.requiredPlugins("Paper");
+		}
+		if (Skript.classExists("io.papermc.paper.event.player.PlayerChangeBeaconEffectEvent")) {
+			Skript.registerEvent("Beacon Change Effect", EvtBeacon.class, PlayerChangeBeaconEffectEvent.class,
+				"beacon change effect", "beacon effect change")
```

Comment:
```suggestion
				"beacon change effect", "beacon effect change", "player chang(e[s]|ing) [of] beacon effect")
```
I don't know about my exact suggestions but the patterns need a few more grammatical ones if possible